src/main/kotlin/org/elaastic/questions/assignment/sequence/interaction/InteractionService.kt

```diff
@@ -36,32 +38,49 @@ import javax.transaction.Transactional
 @Service
 @Transactional
 class InteractionService(
-        @Autowired val interactionRepository: InteractionRepository,
-        @Autowired val sequenceRepository: SequenceRepository,
-        @Autowired val eventLogService: EventLogService,
-        @Autowired val resultsService: ResultsService,
-        @Autowired val responseRepository: ResponseRepository,
-        @Autowired val responseRecommendationService: ResponseRecommendationService
+    @Autowired val interactionRepository: InteractionRepository,
+    @Autowired val sequenceRepository: SequenceRepository,
+    @Autowired val eventLogService: EventLogService,
+    @Autowired val resultsService: ResultsService,
+    @Autowired val responseRepository: ResponseRepository,
+    @Autowired val responseRecommendationService: ResponseRecommendationService
 ) {
 
-    fun create(sequence: Sequence,
-               interactionSpecification: InteractionSpecification,
-               rank: Int,
-               state: State = State.beforeStart): Interaction =
-            Interaction(
-                    interactionType = interactionSpecification.getType(),
-                    rank = rank,
-                    specification = interactionSpecification,
-                    owner = sequence.owner,
-                    sequence = sequence,
-                    state = state
-            ).let(interactionRepository::save)
+    fun create(
+        sequence: Sequence,
+        interactionSpecification: InteractionSpecification,
+        rank: Int,
+        state: State = State.beforeStart
+    ): Interaction =
+        Interaction(
+            interactionType = interactionSpecification.getType(),
+            rank = rank,
+            specification = interactionSpecification,
+            owner = sequence.owner,
+            sequence = sequence,
+            state = state
+        ).let(interactionRepository::save)
 
     fun findById(id: Long): Interaction =
-            interactionRepository.findById(id).get()
+        interactionRepository.findById(id).get()
+
+    fun findResponseByLearnerAssignment(
+        learnerAssignment: LearnerAssignment,
+        sequence: Sequence
+    ): Response? = interactionRepository.findResponseByOwnerAndSequenceAndType(
+        learnerAssignment.learner,
+        sequence,
+        InteractionType.ResponseSubmission
+    )
+
+    fun findAllResponsesBySequenceOrderById(sequence: Sequence): List<Response> =
+        interactionRepository.findAllResponsesBySequenceAndType(
+            sequence,
+            InteractionType.ResponseSubmission
```

Comment:
same comment as above.

Reply:
We must keep `ResponseSubmission` parameter to include it in the JPQL query.
Can we do it another way? We can add a default value to **repository** method definition?

Reply:
Fix bc58dc90e2ee690b9025de3d184ee082bba648b0